src/main/java/com/cloudbees/jenkins/plugins/amazonecs/ECSTaskTemplate.java

```diff
@@ -752,24 +777,69 @@ public ECSTaskTemplate merge(ECSTaskTemplate parent) {
                                      .withHost(hostVolume));
             }
         }
+
+        if (null != efsMountPoints) {
+            for (EFSMountPointEntry mount : efsMountPoints) {
+                if (StringUtils.isEmpty(mount.name)) {
+                    continue;
+                }
+
+                EFSAuthorizationConfig efsAuthorizationConfig = null;
+                if (StringUtils.isNotEmpty(mount.accessPointId)) {
+                    efsAuthorizationConfig = new EFSAuthorizationConfig()
+                            .withAccessPointId(mount.accessPointId)
+                            .withIam(BooleanUtils.isTrue(mount.iam)
+                                    ? EFSAuthorizationConfigIAM.ENABLED
+                                    : EFSAuthorizationConfigIAM.DISABLED
+                            );
+                }
+
+                EFSVolumeConfiguration efsVolumeConfiguration = new EFSVolumeConfiguration()
+                        .withFileSystemId(mount.fileSystemId)
+                        .withRootDirectory(mount.rootDirectory)
+                        .withAuthorizationConfig(efsAuthorizationConfig)
+                        .withTransitEncryption(BooleanUtils.isTrue(mount.transitEncryption)
+                                ? EFSTransitEncryption.ENABLED
+                                : EFSTransitEncryption.DISABLED
+                        );
```

Comment:
We can also extract this as a method called getEFSVolumeConfiguration?
This is to improve readability.